prow/plugins/override/override.go

```diff
@@ -135,6 +155,70 @@ func authorized(gc githubClient, log *logrus.Entry, org, repo, user string) bool
 	return ok
 }
 
+func authorizedTopLevelOwner(oc ownersClient, allowTopLevelOwners bool, log *logrus.Entry, org, repo, user string, pr *github.PullRequest) bool {
+	if allowTopLevelOwners {
+		owners, err := oc.LoadRepoOwners(org, repo, pr.Base.Ref)
+		if err != nil {
+			log.WithError(err).Warnf("cannot determine whether %s is a top level owner of %s/%s", user, org, repo)
+			return false
+		}
+
+		if owners.TopLevelApprovers().Has(user) {
+			return true
+		}
+	}
+	return false
+}
+
+func validateGitHubTeamSlugs(teamSlugs map[string][]string, org, repo string, githubTeams []github.Team) error {
+	slugs := sets.NewString()
+	for _, team := range githubTeams {
+		slugs.Insert(team.Slug)
+	}
+
+	repoRef := fmt.Sprintf("%s/%s", org, repo)
+	invalidSlugs := sets.NewString()
+	for _, slug := range teamSlugs[repoRef] {
+		if !slugs.Has(slug) {
+			invalidSlugs.Insert(slug)
+		}
+	}
+	if invalidSlugs.Len() > 0 {
+		return fmt.Errorf("invalid team slug(s): %s", strings.Join(invalidSlugs.List(), ","))
+	}
+	return nil
+}
+
+func authorizedGitHubTeamMember(gc githubClient, log *logrus.Entry, teamSlugs map[string][]string, org, repo, user string) bool {
+	repoRef := fmt.Sprintf("%s/%s", org, repo)
+	teams, err := gc.ListTeams(org)
+	if err != nil {
+		log.WithError(err).Warnf("cannot get list of teams for org %s", org)
+		return false
+	}
+	if err := validateGitHubTeamSlugs(teamSlugs, org, repo, teams); err != nil {
```

Comment:
This enforces that any teams in the config actually exist? I think it would be OK for me to say `some-people` have rights to `/override` even if at that moment in time the team has not been configured yet -- the authorization check for `/override` would not fail as we'd just not find the team in question, right?

Reply:
Thank you for the review. Yes, the validation only generates a warning to make sure an invalid team slug, e.g. a misspelling of a team's name, does not result in unexpected authorisation failure. I have updated the PR and added a test case to explicitly check for this.

Reply:
I guess my point was that it would not be an auth failure -- it just would not succeed. Sounds good to log at warning level, though.